cmd/snap/cmd_services.go

```diff
@@ -184,8 +185,61 @@ func (s *svcLogs) Execute(args []string) error {
 	return nil
 }
 
+type userAndScopeMixin struct {
+	System bool   `long:"system"`
+	User   bool   `long:"user"`
+	Users  string `long:"users" optional:"yes" optional-value:"all"`
+}
+
+var userAndScopeDescs = mixinDescs{
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"system": i18n.G("The operation should only affect system services."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"user": i18n.G("The operation should only affect user services for the current user."),
+	// TRANSLATORS: This should not start with a lowercase letter.
+	"users": i18n.G("The operation should affect all user services."),
+}
+
+func (um *userAndScopeMixin) validateScopes() error {
+	switch {
```

Comment:
does is make sense to set --system and --user/--users together?

Reply:
As maciej also pointed out, it probably makes sense to allow providing `--system --users` which is equivalent to the default behaviour which is providing nothing. 

I also allow `--system --user` to allow restarting system services and only the current user's services, since they are allowed to use individually, it probably makes sense to just allow them in one go.

OTOH it's a bit confusing so I'm open to modifications on this

Reply:
I disabled `--system --user` together for now, if we think it should be enabled we can add a case in the spread test for it later.